zerver/views/auth.py

```diff
@@ -827,9 +829,37 @@ def api_dev_fetch_api_key(request: HttpRequest, username: str=REQ()) -> HttpResp
         return json_error(_("This user is not registered."),
                           data={"reason": "unregistered"}, status=403)
     do_login(request, user_profile)
-    api_key = get_api_key(user_profile)
+
+    # Return a brand new API key
+    if description is None:
+        # We don't require setting a description to ease development work
+        description = 'Development API key'
+    api_key = create_user_api_key(user_profile, 'Created via REST API').api_key
     return json_success({"api_key": api_key, "email": user_profile.delivery_email})
 
+@csrf_exempt
+@has_request_variables
+def get_user_api_keys(request: HttpRequest, user_profile: UserProfile) -> HttpResponse:
+    """This is for listing all the API keys associated to a user. The keys
+    themselves aren't displayed, just the basic information to identify each
+    one.
+    """
+    api_key_list = (UserAPIKey.objects.filter(user_profile=user_profile)
+                    .values('id', 'description', 'date_created'))
+    return json_success({"api_keys": api_key_list, "email": user_profile.delivery_email})
```

Comment:
```suggestion
    return json_success({"api_keys": api_key_list, "email": user_profile.delivery_email})
```
@YagoGG  in your original commit it was ``` user_profile.email ``` but I changed it to delivery_email, because while merging I found Master HEAD had .delivery_email while sending json response along with api_key. The reason is maybe this commit by ``` 08311566144930f0954bcc7cb39099c77b798847 ``` @timabbott . So is changing .email to .delivery_email fine ? Even though I think this is fine but I just wanted to make sure. 😅

Reply:
Sure, keeping it as `delivery_email` sems to make most sense keeping in mind 08311566144930f0954bcc7cb39099c77b798847.